tests/test_errors.py

```diff
@@ -0,0 +1,131 @@
+import json
+
+from starlette.testclient import TestClient
+
+from aidial_sdk import DIALApp
+from tests.applications.broken_application import BrokenApplication
+from tests.applications.runtime_broken_application import (
+    RuntimeBrokenApplication,
+)
+
+
+def test_error():
+    dial_app = DIALApp()
+    dial_app.add_chat_completion("test_app", BrokenApplication())
+
+    test_app = TestClient(dial_app)
+
+    response = test_app.post(
+        "/openai/deployments/test_app/chat/completions",
+        json={
+            "messages": [{"role": "user", "content": "Test content"}],
+            "stream": False,
+        },
+        headers={"Api-Key": "TEST_API_KEY"},
+    )
+
+    assert response.status_code == 503 and response.json() == {
+        "error": {
+            "message": "Test error",
+            "type": "runtime_error",
+            "code": None,
+            "param": None,
+        }
+    }
+
+
+def test_streaming_error():
+    dial_app = DIALApp()
+    dial_app.add_chat_completion("test_app", BrokenApplication())
+
+    test_app = TestClient(dial_app)
+
+    response = test_app.post(
+        "/openai/deployments/test_app/chat/completions",
+        json={
+            "messages": [{"role": "user", "content": "Test content"}],
+            "stream": True,
+        },
+        headers={"Api-Key": "TEST_API_KEY"},
+    )
+
+    assert response.status_code == 503 and response.json() == {
+        "error": {
+            "message": "Test error",
+            "type": "runtime_error",
+            "code": None,
+            "param": None,
+        }
+    }
+
+
+def test_runtime_streaming_error():
+    dial_app = DIALApp()
+    dial_app.add_chat_completion("test_app", RuntimeBrokenApplication())
+
+    test_app = TestClient(dial_app)
+
+    response = test_app.post(
+        "/openai/deployments/test_app/chat/completions",
+        json={
+            "messages": [{"role": "user", "content": "Test content"}],
+            "stream": True,
+        },
+        headers={"Api-Key": "TEST_API_KEY"},
+    )
+
+    for index, value in enumerate(response.iter_lines()):
```

Comment:
Let's add a helper to deal with streams in tests.
`tests/utils/stream.py: def parse_stream(response) -> List[str]`
which automatically asserts empty lines and returns a list of data chunks without `data: ` prefix.

Reply:
If you don't mind, I suggest doing this as part of https://github.com/epam/ai-dial-sdk/issues/12 (global unit test issue)